src/Themes/ui/ColorPicker/index.tsx

```diff
@@ -0,0 +1,246 @@
+import { ColorPreview } from "./ColorPreview";
+import { HexInput, formatRGBtoHEX } from "./HexInput";
+import { HueSlider } from "./HueSlider";
+import { RGBDigit } from "./RGBDigit";
+import { SVCanvas } from "./SVCanvas";
+import React, { Dispatch, useEffect, useMemo, useReducer, useState } from "react";
+import { useTheme } from "@mui/material/styles";
+import { Modal } from "../../../ui/React/Modal";
+import Button from "@mui/material/Button";
+
+export type HSV = {
+  hue: number;
+  sat: number;
+  val: number;
+};
+
+export type RGB = {
+  r: number;
+  g: number;
+  b: number;
+  a?: number;
+};
+
+function normalize(color: HSV | RGB) {
+  const isRGB = (c: RGB | HSV): c is RGB => Object.hasOwn(color, "r");
+  const rgb: RGB = isRGB(color) ? color : HSVtoRGB(color);
+  const hsv: HSV = !isRGB(color) ? color : RGBtoHSV(color);
+  return { rgb, hsv };
+}
+
+type OpenColorPickerButtonProps = {
+  color: string;
+  title: string;
+  onColorChange: (color: string) => void;
+};
+
+function parseHEXtoRGB(hex: string): RGB {
+  if (hex.length == 4) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[1]}`, 16),
+      g: Number.parseInt(`${hex[2]}${hex[2]}`, 16),
+      b: Number.parseInt(`${hex[3]}${hex[3]}`, 16),
+    };
+  }
+
+  if (hex.length == 7) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[2]}`, 16),
+      g: Number.parseInt(`${hex[3]}${hex[4]}`, 16),
+      b: Number.parseInt(`${hex[5]}${hex[6]}`, 16),
+    };
+  }
+
+  if (hex.length == 9) {
+    return {
+      r: Number.parseInt(`${hex[1]}${hex[2]}`, 16),
+      g: Number.parseInt(`${hex[3]}${hex[4]}`, 16),
+      b: Number.parseInt(`${hex[5]}${hex[6]}`, 16),
+      a: Number.parseInt(`${hex[7]}${hex[8]}`, 16),
+    };
+  }
+
+  throw new Error("Invalid hex string");
+}
+
+export function OpenColorPickerButton({ color, onColorChange }: OpenColorPickerButtonProps) {
```

Comment:
The way this function is named (CapitalCamelCase), and the way it uses state, makes this a function component. That means it should be named accordingly: It is a Thing, not an Action. It should be called ColorPickerButton.

Edit: Perhaps you meant this to be a Thing, as in the Button that Opens the ColorPicker? But "Open" is not really needed here.